fabric-item-api-v1/src/main/java/net/fabricmc/fabric/mixin/item/shears/MatchToolLootConditionMixin.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.item.shears;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.common.collect.ImmutableList;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.item.Item;
+import net.minecraft.item.Items;
+import net.minecraft.loot.condition.LootCondition;
+import net.minecraft.loot.condition.MatchToolLootCondition;
+import net.minecraft.predicate.item.ItemPredicate;
+import net.minecraft.registry.entry.RegistryEntry;
+
+import net.fabricmc.fabric.api.event.lifecycle.v1.CommonLifecycleEvents;
+import net.fabricmc.fabric.impl.item.ShearsHelper;
+import net.fabricmc.fabric.mixin.item.shears.accessors.DirectRegistryEntryListAccessor;
+
+@Mixin(MatchToolLootCondition.class)
+public abstract class MatchToolLootConditionMixin implements LootCondition {
+	@Unique
+	private static final List<ItemPredicate> MATCH_TOOL_PREDICATES = new ArrayList<>();
+
+	@Inject(at = @At("RETURN"), method = "<init>")
+	private void shearsLoot(CallbackInfo ci) {
+		// allows anything in fabric:shears to mine grass (and other stuff) and it will drop
+		((MatchToolLootCondition) (Object) this).predicate().ifPresent(MATCH_TOOL_PREDICATES::add);
+	}
+
+	static {
```

Comment:
Not really sure about using `static` block on Mixin, whether it overwrites the target class' or not. Just to be safe inject to `<clinit>` instead.

Reply:
Mixin `static` blocks are appended at the end of the class's, it is not overwritten. However, this has mostly niche uses and injecting in clinit often works just as well if not better.

Reply:
A static block is in almost all cases preferable to an injection. For example when re-assigning fields you don't need to make them non-final. That doesn't apply here, and they would function the same, but a static block is still much cleaner.

Reply:
so, should i use a static initializer or an inject into <clinit>?
too many opinions lol

Reply:
I wouldnt put this in the static initializer or clinit, as its registering the event handler based on classload.

I would move this to a mod inititliser entrypoint.

Reply:
I think it would be better to not move it into another class, and I also don't think that - registering an event early will do any harm.